src/test/java/uk/sky/cqlmigrate/CqlMigratorImplTest.java

```diff
@@ -152,13 +152,29 @@ public void shouldRemoveLockAfterMigration() throws Exception {
     }
 
     @Test
+    public void shouldRemoveLockAfterMigrationFailed() throws Exception {
+        //given
+        Collection<Path> cqlPaths = singletonList(getResourcePath("cql_bootstrap_missing_semicolon"));
+
+        //when
+        try {
+            MIGRATOR.migrate(CASSANDRA_HOSTS, binaryPort, username, password, TEST_KEYSPACE, cqlPaths);
+        } catch (RuntimeException ignore) {
+        }
+
+        //then
+        ResultSet resultSet = session.execute("SELECT * FROM cqlmigrate.locks WHERE name = ?", LOCK_NAME);
+        assertThat(resultSet.isExhausted()).as("Is lock released").isTrue();
```

Comment:
In our case, we don't proceed with deployment of the second node, if the first node fails. In that case, the failure of the deployment is already a matter for investigation, so we don't need another manual action to do something if the problem is explained in logs, giving us a hint to check the database.

So, maybe it needs to be configurable?